sjsonnet/src/sjsonnet/Interpreter.scala

```diff
@@ -107,7 +107,7 @@ class Interpreter(extVars: Map[String, String],
             override def evalDefault(expr: Expr, vs: ValScope, es: EvalScope) = {
               evaluator.visitExpr(expr)(if (tlaExpressions.exists(_ eq expr)) ValScope.empty else vs)
             }
-          }
+          }.apply0(f.pos)(evaluator)
```

Comment:
The old code was calling

```
apply(Evaluator.emptyLazyArray, null, outerPos)
```

but this new code is calling

```scala
    def apply0(outerPos: Position)(implicit ev: EvalScope): Val = {
      if(params.names.length != 0) apply(Evaluator.emptyLazyArray, null, outerPos)
      else {
        val funDefFileScope: FileScope = pos match { case null => outerPos.fileScope case p => p.fileScope }
        evalRhs(defSiteValScope, ev, funDefFileScope, outerPos)
      }
    }
```

and I think these are evidently equivalent in case the function has parameters.

In the case where the function _doesn't_ have parameters, it looks like it goes into an `evalRhs` path but that just looks like an optimization to short-circuit parts of `apply()` (i.e. it corresponds to the `if(simple)` branch in `apply` but without having to generate garbage to handle args (which aren't present)).

Therefore I think these are behaviorally equivalent 👍